lib/charms/vault_k8s/v0/vault_tls.py

```diff
@@ -99,7 +130,8 @@ def __init__(self, charm, peer_relation: str, substrate: Substrate):
             self._on_tls_certificates_access_relation_broken,
         )
         self.framework.observe(
-            self.tls_access.on.certificate_available, self._on_certificate_config_changed
+            self.tls_access.on.certificate_available,
+            self._on_tls_certificates_access_certificate_available,
```

Comment:
We have decided against listening to this event and centralizing the logic in the configure function in the first implementation of this library, so this should be reversed.

Reply:
Done.